src/main/scala/tech/cryptonomic/conseil/tezos/PlatformDiscoveryOperations.scala

```diff
@@ -0,0 +1,93 @@
+package tech.cryptonomic.conseil.tezos
+
+import com.typesafe.config.Config
+import slick.ast.FieldSymbol
+import slick.jdbc.PostgresProfile
+import slick.jdbc.PostgresProfile.api._
+import tech.cryptonomic.conseil.tezos.PlatformDiscoveryTypes.DataType.DataType
+import tech.cryptonomic.conseil.tezos.PlatformDiscoveryTypes._
+import tech.cryptonomic.conseil.tezos.{TezosDatabaseOperations => TezosDb}
+import tech.cryptonomic.conseil.util.DatabaseUtil
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+
+object PlatformDiscoveryOperations {
+
+  lazy val dbHandle: PostgresProfile.backend.Database = DatabaseUtil.db
+
+  private val tables = List(Tables.Blocks, Tables.Accounts, Tables.OperationGroups, Tables.Operations, Tables.Fees)
+  private val tablesMap = tables.map(table => table.baseTableRow.tableName -> table)
+
+  def getNetworks(config: Config): List[Network] = {
+    config.getObject("platforms").asScala.flatMap {
+      case (platform, strippedConf) =>
+        strippedConf.atKey(platform).getObject(platform).asScala.map {
+          case (network, _) =>
+            Network(network, network.capitalize, platform, network)
+        }.toList
+    }.toList
+  }
+
+  def getEntities(network: String)(implicit ec: ExecutionContext): Future[List[Entity]] = {
+    ApiOperations.countAll.map { counts =>
+      tables.map(_.baseTableRow.tableName).flatMap { tableName =>
```

Comment:
This could now be written as `tablesMap.keys.flatMap` or equivalent `tablesMap.keySet.flatMap`

Reply:
Unfortunately it is not a real `Map` and I cannot create one -> I've got an error from compiler:
```
PlatformDiscoveryOperations.scala:21:86: Cannot prove that (String, slick.lifted.TableQuery[_1])( forSome { type _1 >: tech.cryptonomic.conseil.tezos.Tables.Fees with tech.cryptonomic.conseil.tezos.Tables.Operations with tech.cryptonomic.conseil.tezos.Tables.OperationGroups with tech.cryptonomic.conseil.tezos.Tables.Accounts with tech.cryptonomic.conseil.tezos.Tables.Blocks <: tech.cryptonomic.conseil.tezos.Tables.profile.Table[_ >: tech.cryptonomic.conseil.tezos.Tables.FeesRow with tech.cryptonomic.conseil.tezos.Tables.OperationsRow with tech.cryptonomic.conseil.tezos.Tables.OperationGroupsRow with tech.cryptonomic.conseil.tezos.Tables.AccountsRow with tech.cryptonomic.conseil.tezos.Tables.BlocksRow <: Product with Serializable] }) <:< (T, U).
[error]   private val tablesMap = tables.map(table => table.baseTableRow.tableName -> table).toMap
```
I changed it but not the way you suggested

Reply:
Fine anyway!

Just FYI, I guess you simply could have defined `tablesMap` ending with `.toMap`? To have a real scala.Map

Reply:
check out the error I've pasted above,`.toMap` doesn't work, I guess because I have a bit more extended `TableQuery` - apparently if I had typed it explicitly as `TableQuery` it would not infer any of methods I use here.